test/e2e/e2e.go

```diff
@@ -742,6 +748,504 @@ var _ = Describe("validation", func() {
 			}, timeout, interval).Should(BeTrue())
 		})
 	})
+
+	Context("Creating BGP Peer", func() {
+		table.DescribeTable("Testing creating BGP peer CR successfully", func(peerName string, peer *metallbv1alpha1.BGPPeer, expectedConfigMap string) {
+			By("By creating BGP peer CR")
+
+			Expect(testclient.Client.Create(context.Background(), peer)).Should(Succeed())
+
+			key := types.NamespacedName{
+				Name:      peerName,
+				Namespace: OperatorNameSpace,
+			}
+			// Create BGP Peer resource
+			By("By checking BGP peer resource is created")
+			Eventually(func() error {
+				err := testclient.Client.Get(context.Background(), key, peer)
+				return err
+			}, timeout, interval).Should(Succeed())
+
+			// Checking ConfigMap is created
+			By("By checking ConfigMap is created match the expected configuration")
+			Eventually(func() (string, error) {
+				configmap, err := testclient.Client.ConfigMaps(OperatorNameSpace).Get(context.Background(), consts.MetalLBConfigMapName, metav1.GetOptions{})
+				if err != nil {
+					return "", err
+				}
+				return configmap.Data[consts.MetalLBConfigMapName], err
+			}, timeout, interval).Should(MatchYAML(expectedConfigMap))
+
+			By("By checking BGP Peer resource and ConfigMap are deleted")
+			Eventually(func() bool {
+				err := testclient.Client.Delete(context.Background(), peer)
```

Comment:
might be better delete the resource once (before the eventually) and then inside the eventually use Get to check if the error is NotFound.